Week-4/session-11/oopsEs6/ abstractionAndEncpasulation.js

```diff
@@ -0,0 +1,59 @@
+class BankAccount {
+  #accountNumber;
+  #balance;
+  #accountHolderName;
+
+  constructor(accountNumber, balance, accountHolderName) {
+    this.#accountNumber = accountNumber;
+    this.#balance = balance;
+    this.#accountHolderName = accountHolderName;
+  }
+
+  deposit(amount) {
```

Comment:
The deposit() and withdraw() methods are defined in the BankAccount superclass, which is not necessary as per your requirements. These methods should be implemented in the subclasses CheckingAccount and SavingsAccount to adhere to the guidelines.